translate library/weakref.po

ken71301 opened the pull request.

ken71301 commented:
ready for review~

mattwang44 commented on library/weakref.po:
```diff
@@ -30,13 +31,14 @@ msgstr "**原始碼：**\\ :source:`Lib/weakref.py`"
 "The :mod:`weakref` module allows the Python programmer to create :dfn:`weak "
 "references` to objects."
-msgstr ""
+msgstr ":mod:`weakref` 模組允許 Python 程式設計師建立對物件的 :dfn:`弱參照`。"
```
沒深究過 dfn 是什麼作用，但以前我都是比照其他 role 處理XD 可以幫忙確認一下嗎？
```suggestion
msgstr ":mod:`weakref` 模組允許 Python 程式設計師建立對物件的\\ :dfn:`弱參照 <weak references>`。"
```

ken71301 replied:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-dfn
看起來是一樣

mattwang44 replied:
> No index entries are generated.

加上 doc build script 也不會像是其他 role 去比對原文譯文，實質上應該只有標示作用而已，那應該維持現狀就好

rockleona commented on library/weakref.po:
```diff
@@ -136,6 +173,9 @@ msgid ""
 "of strings in the ``__slots__`` declaration. See :ref:`__slots__ "
 "documentation <slots>` for details."
 msgstr ""
+"當為給定的型別定義 ``__slots__`` 時，弱參照支援將被停用，除非 "
+"``'__weakref__'`` 字串也存在於 ``__slots__`` 宣告的字串序列中。詳情請參閱 :"
+"ref:`__slots__ 文件<slots>`。"
```
這邊應該要有空格吧？

```suggestion
"當為給定的型別定義 ``__slots__`` 時，弱參照支援將被停用，除非 "
"``'__weakref__'`` 字串也存在於 ``__slots__`` 宣告的字串序列中。詳情請參閱 :"
"ref:`__slots__ 文件 <slots>`。"
```

rockleona reviewed (commented):
抓時間 review 到 L116

rockleona commented on library/weakref.po:
```diff
@@ -170,6 +219,9 @@ msgid ""
 "`hash` is called the first time only after the *object* was deleted, the "
 "call will raise :exc:`TypeError`."
 msgstr ""
+"如果 *object* 是可雜湊的，那麼弱參照就是 :term:`hashable`。即使在 *object* 被"
+"刪除後，它們仍將保留其雜湊值。如果僅在 *object* 刪除後 :func:`hash` 才第一次"
+"被呼叫，則該呼叫將引發 :exc:`TypeError`。"
```
hmmmmm, 這句好難呀，不知道這樣會不會比較通順一點？
只怕偏離原意@@

```suggestion
"如果 *object* 是可雜湊的，那麼弱參照就是 :term:`hashable`。即使在 *object* 被"
"刪除後，它們仍將保留其雜湊值。如果僅在 *object* 刪除後才第一次"
"呼叫 :func:`hash`，則該呼叫將引發 :exc:`TypeError`。"
```

rockleona commented on library/weakref.po:
```diff
@@ -207,29 +264,35 @@ msgid ""
 "keys.  *callback* is the same as the parameter of the same name to the :func:"
 "`ref` function."
 msgstr ""
+"傳回一個使用弱參照的 *object* 的代理 (proxy)。這支援在大多數情境中使用代理，"
+"而不需要對弱參照物件明確地取消參照。回傳的物件將具有 ``ProxyType`` 或 "
+"``CallableProxyType`` 型別，具體取決於 *object* 是否為可呼叫物件。無論參照目"
+"標如何，代理物件都不 :term:`hashable`；這避免了與其基本可變物件本質相關的許多"
+"問題，並阻止它們作為字典的鍵被使用。*callback* 與 :func:`ref` 函式的同名參數"
+"是相同的。"
```
nit

```suggestion
"傳回一個使用弱參照的 *object* 的代理 (proxy)。這支援在大多數情境中使用代理，"
"而不需要對弱參照物件明確地取消參照。回傳的物件將具有 ``ProxyType`` 或 "
"``CallableProxyType`` 型別，具體取決於 *object* 是否為可呼叫物件。無論參照目"
"標如何，代理物件都不是 :term:`hashable`；這避免了與其基本可變物件本質相關的許多"
"問題，並阻止它們作為字典的鍵被使用。*callback* 與 :func:`ref` 函式的同名參數"
"是相同的。"
```

rockleona commented on library/weakref.po:
```diff
@@ -247,10 +313,13 @@ msgid ""
 #: ../../library/weakref.rst:188
 msgid "A workaround would be to remove the key prior to reassignment::"
-msgstr ""
+msgstr "解決方法是在重新賦值 (reassignment) 之前刪除鍵： ::"
```
nit，但我覺得是個有趣的議題 XD
`workaround`： 比較像是繞個彎解決問題的方法
`solution`： 真正的解決問題的方法

我覺得值得討論一下怎麼翻譯這裡

```suggestion
msgstr "變通方法是在重新賦值 (reassignment) 之前刪除鍵： ::"
```

ken71301 replied:
或 一個變通的解法...好像也行

rockleona replied:
聽起來也不錯！

ken71301 replied:
```suggestion
msgstr "一個變通的解法是在重新賦值 (reassignment) 之前刪除鍵： ::"
```

rockleona reviewed (commented):
Finished Reivew

ken71301 committed:
fix: Inline literal start-string without end-string.

feat: to rst:264

Update library/weakref.po

Co-authored-by: RockLeon <34214497+rockleona@users.noreply.github.com>

Apply suggestions from code review

Co-authored-by: RockLeon <34214497+rockleona@users.noreply.github.com>

Apply suggestions from code review

Co-authored-by: Wei-Hsiang (Matt) Wang <mattwang44@gmail.com>

fix: Apply suggestions from code review

Apply suggestions from code review

Co-authored-by: Wei-Hsiang (Matt) Wang <mattwang44@gmail.com>

mattwang44 commented on library/weakref.po:
```diff
@@ -106,28 +133,38 @@ msgid ""
 msgstr ""
+"並非所有物件都可以被弱參照。支援弱參照的物件包括類別實例、用 Python （但不是"
+"C）編寫的函式、實例方法、集合、凍結集合 (frozenset)、一些\\ :term:`檔案物件 "
+"<file object>`、:term:`產生器 <generator>`、類型物件、插座 (socket)、陣列、雙"
```
nit: socket 應該直接原文就好，之前討論印象中是任何相關翻譯都不通用

ken71301 replied:
ok
```suggestion
"<file object>`、:term:`產生器 <generator>`、類型物件、socket、陣列、雙"
```

mattwang44 commented on library/weakref.po:
```diff
@@ -170,6 +219,9 @@ msgid ""
 "`hash` is called the first time only after the *object* was deleted, the "
 "call will raise :exc:`TypeError`."
 msgstr ""
+"如果 *object* 是可雜湊的，那麼弱參照就是 :term:`hashable`。即使在 *object* 被"
```
```suggestion
"如果 *object* 是\\ :term:`可雜湊的 <hashable>`，那麼弱參照就是可雜湊的。即使在 *object* 被"
```

mattwang44 commented on library/weakref.po:
```diff
@@ -305,12 +381,16 @@ msgid ""
 "hold of it.  :class:`WeakMethod` has special code to recreate the bound "
 "method until either the object or the original function dies::"
 msgstr ""
+"一個特製的 :class:`ref` 子類別，其模擬對繫結函式 (bound method ) （即在類別上"
```
method -> 方法？
```suggestion
"一個特製的 :class:`ref` 子類別，其模擬對繫結方法 (bound method) （即在類別上"
```

ken71301 replied:
oops

mattwang44 replied:
該句有數個要修改的地方噢

mattwang44 commented on library/weakref.po:
```diff
@@ -305,12 +381,16 @@ msgid ""
 msgstr ""
+"一個特製的 :class:`ref` 子類別，其模擬對繫結方法 (bound method) （即在類別上"
+"定義並在實例上查找的方法）的弱參照。由於繫結方法是短暫存在的，因此標準弱參照"
+"無法保留它。:class:`WeakMethod` 有特殊的程式碼來重新建立繫結函式，直到物件或"
```
```suggestion
"無法保留它。:class:`WeakMethod` 有特殊的程式碼來重新建立繫結方法，直到物件或"
```
